Task2 solved but without tests

velfor opened the pull request:
Task 2
Add a new entity Review: 
id (autogenerated)
status (enum: DRAFT, CANCELLED, DONE)
employee_id
created_at (date and time of record creation, not null)
 
In Employee, add a <Review> OneToMany collection (one employee has many reviews).
 
Add api returning only employee with is_active = true
The fields in the response should be only the following and with these names:  
1. id;
    2. firstName;
    3. surname; (last_name field value)
    4. email;
	5. reviews collection (id, status, employeeId, createdAt)
 
Ability to sort by firstName, surname and email. In descending, ascending order. Paging.
Add tests for the controller.

The task was given out on 18 March 2024, Uladzimir Kuzniatsou
The task was completed on 25 March 2024, Pavel Nikiforov

velfor committed:
Review, EmployeeDto, EmployeeMapper were added

velfor committed:
Task2 solved but without tests

velfor committed:
Readme.md with information about end-points was added

UladzimirKuzniatsou reviewed (approved):
let it be. will discuss the code.

Commit:
Another one pageable API added for Employee get

Commit:
Another one pageable API added for Employee get

velfor committed:
Controller tests were added

velfor committed:
Validation added

velfor committed:
Controller methods now accept EmployeeDto instead of Employee.
Added Review <-> ReviewDto, List<Review> <-> List<ReviewDto> mappers for correct Employee <-> EmployeeDto mapping.
Validation takes place at the controller parameter level.

velfor committed:
All tests  have been fixed, all tests pass
